### PR TITLE
[layout] manual layout should increment revision number

### LINKED ISSUE #23: [layout] manual layout should increment revision number
To be able to detect on the client side that a manual layout is a change to the current model, we should increment the revision number on manual layout actions.

### PR DESCRIPTION
**Pull request by JanKoehnlein**
Fixes #23

### CODE REVIEW

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java [hunk 1]
```diff
@@ -455,10 +455,20 @@ protected void handle(LayoutAction action) {
...
+				}
+				layoutEngine.layout(newRoot);
+				doSubmitModel(newRoot, true);
 			}
 		}
-		doSubmitModel(currentRoot, true);
+	}
+	
+	protected SModelCloner getSModelCloner() {
+		return new SModelCloner();
```
Comment (spoenemann) [inline org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: Should we inject this?

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java [hunk 2]
```diff
@@ -455,10 +455,20 @@ protected void handle(LayoutAction action) {
 		if (getServerLayoutKind(currentRoot) == ServerLayoutKind.MANUAL) {
 			ILayoutEngine layoutEngine = getLayoutEngine();
 			if (layoutEngine != null) {
-				layoutEngine.layout(currentRoot);
+				SModelCloner cloner = getSModelCloner();
+				SModelRoot newRoot = cloner.clone(currentRoot);
```
Comment (spoenemann) [inline org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: Could you add a comment on why cloning is necessary here?

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java (hunk 1 above)
Comment (JanKoehnlein) [reply org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: OK

File: org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java (hunk 2 above)
Comment (JanKoehnlein) [reply org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java]: Added

**Pull request merged**